multicluster/service-mirror/cluster_watcher.go

```diff
@@ -349,29 +375,56 @@ func (rcsw *RemoteClusterServiceWatcher) cleanupMirroredResources(ctx context.Co
 		}
 	}
 
-	endpoints, err := rcsw.localAPIClient.Endpoint().Lister().List(labels.Set(matchLabels).AsSelector())
-	if err != nil {
-		innerErr := fmt.Errorf("could not retrieve endpoints that need cleaning up: %s", err)
-		if kerrors.IsNotFound(err) {
-			return innerErr
+	if rcsw.enableEndpointSlices {
+		endpoints, err := rcsw.localAPIClient.ES().Lister().List(labels.Set(matchLabels).AsSelector())
+		if err != nil {
+			innerErr := fmt.Errorf("could not retrieve endpoints that need cleaning up: %s", err)
```

Comment:
It may be worth differentiating this error from the endpoints error so that it's easier to see what we're watching

```suggestion
			innerErr := fmt.Errorf("could not retrieve EndpointSlices that need cleaning up: %s", err)
```